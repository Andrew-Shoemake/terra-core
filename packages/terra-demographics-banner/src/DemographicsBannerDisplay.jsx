import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl, intlShape } from 'react-intl';
import ResponsiveElement from 'terra-responsive-element';
import './DemographicsBanner.module.scss';
import SmallDemographicsBannerDisplay from './_SmallDemographicsBannerDisplay';
import LargeDemographicsBannerDisplay from './_LargeDemographicsBannerDisplay';

const identifierShape = {
  /**
   * The label of the identifier: e.g.: `SSN`
   */
  label: PropTypes.string,
  /**
   * The value to display for the identifier. e.g.: `111-22-3333`
   */
  value: PropTypes.string,
  /**
   * The aria-label for the identifier. e.g.: `Social Security Number 1 1 1 2 2 3 3 3 3`
   */
  ariaLabel: PropTypes.string,
};

const stringShape = {
  /**
   * The value to display. e.g.: `03/15/2020`
   */
  value: PropTypes.string,
  /**
   * The aria-label for the value. e.g.: `March 15th, 2020`
   */
  ariaLabel: PropTypes.string,
};

const administrativeSexShape = {
  /**
   * The sex to display. e.g.: `M`
   */
  value: PropTypes.string,
  /**
   * The aria-label for the sex. e.g.: `Male`
   */
  ariaLabel: PropTypes.string,
  /**
   * Whether or not the administrative sex is different from the birth sex. If different, an icon will be displayed to denote the mismatch in sex.
   */
  isBirthSexMismatch: PropTypes.bool,
};

const propTypes = {
  /**
   * Administrative sex of the Person. This will be displayed on the demographics banner.
   */
  administrativeSex: PropTypes.shape(administrativeSexShape),
  /**
   * Text description of the Administrative Sex label that is read by screen readers
   */
  administrativeSexFullText: PropTypes.string,
  /**
   * Age of the person.
   */
  age: PropTypes.shape(stringShape),
  /**
   * Application content to display in the banner.
   */
  applicationContent: PropTypes.node,
  /**
   * The persons date of birth
   */
  dateOfBirth: PropTypes.shape(stringShape),
  /**
   * Label to display for the date of birth
   */
  dateOfBirthLabel: PropTypes.string,
  /**
   * Text Description of the date of birth label that is read by screen readers
   */
  dateOfBirthFullText: PropTypes.string,
  /**
   * The person's deceased date. Will display the banner as deceased if this value is provided
   */
  deceasedDate: PropTypes.shape(stringShape),
  /**
   * Label to display for the deceased date
   */
  deceasedDateLabel: PropTypes.string,
  /**
   * The persons gestational age.
   */
  gestationalAge: PropTypes.shape(stringShape),
  /**
   * Label to display for the Gestational Age
   */
  gestationalAgeLabel: PropTypes.string,
  /**
  * Text Description of the Gestational Age label that is read by screen readers
   */
  gestationalAgeFullText: PropTypes.string,
  /**
   * Additional array of label value identifiers of a person's demographic information.
   */
  identifiers: PropTypes.arrayOf(PropTypes.shape(identifierShape)),
  /**
   * @private
   * The intl object containing translations. This is retrieved from the context automatically by injectIntl.
   */
  intl: intlShape.isRequired,
  /**
   * @private
   * Whether or not the banner should show active styles
   */
  isActive: PropTypes.bool,
  /**
   * Whether to display an icon to confidentiality.
   */
  isConfidential: PropTypes.bool,
  /**
   * @private
   * Whether or not the banner should have additional styles and to indicate it is selectable.
   */
  isSelectable: PropTypes.bool,
  /**
   * Full Name of the person
   */
  personName: PropTypes.string,
  /**
   * Photo to display in the banner
   */
  photo: PropTypes.node,
  /**
   * The person's postmenstrual age.
   */
  postMenstrualAge: PropTypes.shape(stringShape),
  /**
   * Label to display for the PostMenstrualAgeLabel
   */
  postMenstrualAgeLabel: PropTypes.string,
  /**
   * Text description of the Post Menstrural Age label that is read by screen readers
   */
  postMenstrualAgeFullText: PropTypes.string,
  /**
   * The persons preferred first name if they have one.
   */
  preferredFirstName: PropTypes.string,
  /**
   * Text description that is read by screen readers when the Administrative Sex and Birth Sex are not identical.
   */
  sexMismatchFullText: PropTypes.string,
};

const defaultProps = {
  administrativeSex: {},
  administrativeSexFullText: '',
  age: {},
  applicationContent: null,
  dateOfBirth: {},
  dateOfBirthLabel: '',
  dateOfBirthFullText: '',
  deceasedDate: {},
  deceasedDateLabel: '',
  gestationalAge: {},
  gestationalAgeLabel: '',
  gestationalAgeFullText: '',
  isActive: false,
  isConfidential: false,
  isSelectable: false,
  personName: '',
  postMenstrualAge: {},
  postMenstrualAgeLabel: '',
  postMenstrualAgeFullText: '',
  preferredFirstName: null,
  sexMismatchFullText: '',
};

const DemographicsBannerDisplay = props => (
  <ResponsiveElement
    tiny={<SmallDemographicsBannerDisplay {...props} />}
    small={<LargeDemographicsBannerDisplay {...props} />}
  />
);

DemographicsBannerDisplay.propTypes = propTypes;
DemographicsBannerDisplay.defaultProps = defaultProps;

export default injectIntl(DemographicsBannerDisplay);
