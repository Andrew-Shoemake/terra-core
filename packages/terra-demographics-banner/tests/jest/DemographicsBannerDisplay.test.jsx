import React from 'react';
/* eslint-disable-next-line import/no-extraneous-dependencies */
import { shallowWithIntl } from 'terra-enzyme-intl';
import DemographicsBannerDisplay from '../../src/DemographicsBannerDisplay';

it('renders a blank banner properly', () => {
  const banner = shallowWithIntl(<DemographicsBannerDisplay />);
  expect(banner).toMatchSnapshot();
});

it('renders a banner that has basic information', () => {
  const banner = shallowWithIntl(<DemographicsBannerDisplay
    age="24"
    dateOfBirth="March 30, 2017"
    gender="F"
    personName="Terra Demographics Banner"
  />);

  expect(banner).toMatchSnapshot();
});

it('renders the banner that contains additional information', () => {
  const banner = shallowWithIntl(<DemographicsBannerDisplay
    applicationContent={<span className="risk-score">5%</span>}
    age="25 Years"
    dateOfBirth="May 9, 1993"
    gender="Male"
    identifiers={{ MRN: 12343, REA: '3JSDA' }}
    personName="Johnathon Doe"
    preferredFirstName="John"
  />);

  expect(banner).toMatchSnapshot();
});

it('renders the banner appropriately for a deceased person', () => {
  const banner = shallowWithIntl(<DemographicsBannerDisplay
    age="24"
    gender="F"
    dateOfBirth="June 15, 2014"
    deceasedDate="March 30, 2017"
    personName="Old Terra Demographics Banner"
  />);

  expect(banner).toMatchSnapshot();
});

it('renders the banner appropriately for a person with gestational and post menstrual dates', () => {
  const banner = shallowWithIntl(<DemographicsBannerDisplay
    applicationContent={<span className="risk-score">5%</span>}
    age="25 Years"
    dateOfBirth="May 9, 1993"
    gender="Male"
    gestationalAge="April 5, 2016"
    gestationalAgeLabel="GA"
    identifiers={{ MRN: 12343, REA: '3JSDA' }}
    personName="Johnathon Doe"
    postMenstrualAge="April 7, 2016"
    postMenstrualAgeLabel="PMA"
    preferredFirstName="John"
  />);

  expect(banner).toMatchSnapshot();
});

it('renders gestational date with a default label when the label is not provided', () => {
  const banner = shallowWithIntl(<DemographicsBannerDisplay gestationalAge="April 5, 2016" />);
  expect(banner).toMatchSnapshot();
});

it('renders post menstrural date with a default label when the label is not provided', () => {
  const banner = shallowWithIntl(<DemographicsBannerDisplay postMenstrualAge="April 5, 2016" />);
  expect(banner).toMatchSnapshot();
});

it('renders the banner properly for a deceased person with additional application content', () => {
  const banner = shallowWithIntl(<DemographicsBannerDisplay
    applicationContent={<span className="risk-score">5%</span>}
    age="25 Years"
    dateOfBirth="May 9, 1993"
    deceasedDate="March 5, 2016"
    gender="Male"
    identifiers={{ MRN: 12343, REA: '3JSDA' }}
    personName="Johnathon Doe"
    preferredFirstName="John"
  />);

  expect(banner).toMatchSnapshot();
});

it('renders the banner with avatar', () => {
  const banner = shallowWithIntl(<DemographicsBannerDisplay
    applicationContent={<span className="risk-score">5%</span>}
    age="25 Years"
    avatar={{
      alt: 'Johnathon Doe',
      color: 'neutral',
      hashValue: 'alternative hash',
      image: 'avatar',
      initials: 'JD',
      isAriaHidden: true,
    }}
    dateOfBirth="May 9, 1993"
    gender="Male"
    identifiers={{ MRN: 12343, REA: '3JSDA' }}
    personName="Johnathon Doe"
    preferredFirstName="John"
  />);

  expect(banner).toMatchSnapshot();
});

it('renders the banner with confidential icon', () => {
  const banner = shallowWithIntl(<DemographicsBannerDisplay
    applicationContent={<span className="risk-score">5%</span>}
    age="25 Years"
    dateOfBirth="May 9, 1993"
    gender="Male"
    identifiers={{ MRN: 12343, REA: '3JSDA' }}
    isConfidential
    personName="Johnathon Doe"
    preferredFirstName="John"
  />);

  expect(banner).toMatchSnapshot();
});
