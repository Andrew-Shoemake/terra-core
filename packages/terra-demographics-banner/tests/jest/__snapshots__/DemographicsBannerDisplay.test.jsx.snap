// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a banner that has basic information 1`] = `
<ResponsiveElement
  responsiveTo="parent"
  small={
    <_default
      age="24"
      applicationContent={null}
      dateOfBirth="March 30, 2017"
      dateOfBirthFullText="Date of Birth"
      dateOfBirthLabel="DOB"
      deceasedDate={null}
      deceasedDateLabel="Deceased"
      gender="F"
      gestationalAge={null}
      gestationalAgeFullText="Gestational Age"
      gestationalAgeLabel="GA"
      identifiers={Object {}}
      isConfidential={false}
      personName="Terra Demographics Banner"
      postMenstrualAge={null}
      postMenstrualAgeFullText="Post Menstrual Age"
      postMenstrualAgeLabel="PMA"
      preferredFirstName={null}
    />
  }
  tiny={
    <_default
      age="24"
      applicationContent={null}
      dateOfBirth="March 30, 2017"
      dateOfBirthFullText="Date of Birth"
      dateOfBirthLabel="DOB"
      deceasedDate={null}
      deceasedDateLabel="Deceased"
      gender="F"
      gestationalAge={null}
      gestationalAgeFullText="Gestational Age"
      gestationalAgeLabel="GA"
      identifiers={Object {}}
      isConfidential={false}
      personName="Terra Demographics Banner"
      postMenstrualAge={null}
      postMenstrualAgeFullText="Post Menstrual Age"
      postMenstrualAgeLabel="PMA"
      preferredFirstName={null}
    />
  }
/>
`;

exports[`renders a blank banner properly 1`] = `
<ResponsiveElement
  responsiveTo="parent"
  small={
    <_default
      age="--"
      applicationContent={null}
      dateOfBirth="--"
      dateOfBirthFullText="Date of Birth"
      dateOfBirthLabel="DOB"
      deceasedDate={null}
      deceasedDateLabel="Deceased"
      gender="--"
      gestationalAge={null}
      gestationalAgeFullText="Gestational Age"
      gestationalAgeLabel="GA"
      identifiers={Object {}}
      isConfidential={false}
      personName="--"
      postMenstrualAge={null}
      postMenstrualAgeFullText="Post Menstrual Age"
      postMenstrualAgeLabel="PMA"
      preferredFirstName={null}
    />
  }
  tiny={
    <_default
      age="--"
      applicationContent={null}
      dateOfBirth="--"
      dateOfBirthFullText="Date of Birth"
      dateOfBirthLabel="DOB"
      deceasedDate={null}
      deceasedDateLabel="Deceased"
      gender="--"
      gestationalAge={null}
      gestationalAgeFullText="Gestational Age"
      gestationalAgeLabel="GA"
      identifiers={Object {}}
      isConfidential={false}
      personName="--"
      postMenstrualAge={null}
      postMenstrualAgeFullText="Post Menstrual Age"
      postMenstrualAgeLabel="PMA"
      preferredFirstName={null}
    />
  }
/>
`;

exports[`renders gestational date with a default label when the label is not provided 1`] = `
<ResponsiveElement
  responsiveTo="parent"
  small={
    <_default
      age="--"
      applicationContent={null}
      dateOfBirth="--"
      dateOfBirthFullText="Date of Birth"
      dateOfBirthLabel="DOB"
      deceasedDate={null}
      deceasedDateLabel="Deceased"
      gender="--"
      gestationalAge="April 5, 2016"
      gestationalAgeFullText="Gestational Age"
      gestationalAgeLabel="GA"
      identifiers={Object {}}
      isConfidential={false}
      personName="--"
      postMenstrualAge={null}
      postMenstrualAgeFullText="Post Menstrual Age"
      postMenstrualAgeLabel="PMA"
      preferredFirstName={null}
    />
  }
  tiny={
    <_default
      age="--"
      applicationContent={null}
      dateOfBirth="--"
      dateOfBirthFullText="Date of Birth"
      dateOfBirthLabel="DOB"
      deceasedDate={null}
      deceasedDateLabel="Deceased"
      gender="--"
      gestationalAge="April 5, 2016"
      gestationalAgeFullText="Gestational Age"
      gestationalAgeLabel="GA"
      identifiers={Object {}}
      isConfidential={false}
      personName="--"
      postMenstrualAge={null}
      postMenstrualAgeFullText="Post Menstrual Age"
      postMenstrualAgeLabel="PMA"
      preferredFirstName={null}
    />
  }
/>
`;

exports[`renders post menstrural date with a default label when the label is not provided 1`] = `
<ResponsiveElement
  responsiveTo="parent"
  small={
    <_default
      age="--"
      applicationContent={null}
      dateOfBirth="--"
      dateOfBirthFullText="Date of Birth"
      dateOfBirthLabel="DOB"
      deceasedDate={null}
      deceasedDateLabel="Deceased"
      gender="--"
      gestationalAge={null}
      gestationalAgeFullText="Gestational Age"
      gestationalAgeLabel="GA"
      identifiers={Object {}}
      isConfidential={false}
      personName="--"
      postMenstrualAge="April 5, 2016"
      postMenstrualAgeFullText="Post Menstrual Age"
      postMenstrualAgeLabel="PMA"
      preferredFirstName={null}
    />
  }
  tiny={
    <_default
      age="--"
      applicationContent={null}
      dateOfBirth="--"
      dateOfBirthFullText="Date of Birth"
      dateOfBirthLabel="DOB"
      deceasedDate={null}
      deceasedDateLabel="Deceased"
      gender="--"
      gestationalAge={null}
      gestationalAgeFullText="Gestational Age"
      gestationalAgeLabel="GA"
      identifiers={Object {}}
      isConfidential={false}
      personName="--"
      postMenstrualAge="April 5, 2016"
      postMenstrualAgeFullText="Post Menstrual Age"
      postMenstrualAgeLabel="PMA"
      preferredFirstName={null}
    />
  }
/>
`;

exports[`renders the banner appropriately for a deceased person 1`] = `
<ResponsiveElement
  responsiveTo="parent"
  small={
    <_default
      age="24"
      applicationContent={null}
      dateOfBirth="June 15, 2014"
      dateOfBirthFullText="Date of Birth"
      dateOfBirthLabel="DOB"
      deceasedDate="March 30, 2017"
      deceasedDateLabel="Deceased"
      gender="F"
      gestationalAge={null}
      gestationalAgeFullText="Gestational Age"
      gestationalAgeLabel="GA"
      identifiers={Object {}}
      isConfidential={false}
      personName="Old Terra Demographics Banner"
      postMenstrualAge={null}
      postMenstrualAgeFullText="Post Menstrual Age"
      postMenstrualAgeLabel="PMA"
      preferredFirstName={null}
    />
  }
  tiny={
    <_default
      age="24"
      applicationContent={null}
      dateOfBirth="June 15, 2014"
      dateOfBirthFullText="Date of Birth"
      dateOfBirthLabel="DOB"
      deceasedDate="March 30, 2017"
      deceasedDateLabel="Deceased"
      gender="F"
      gestationalAge={null}
      gestationalAgeFullText="Gestational Age"
      gestationalAgeLabel="GA"
      identifiers={Object {}}
      isConfidential={false}
      personName="Old Terra Demographics Banner"
      postMenstrualAge={null}
      postMenstrualAgeFullText="Post Menstrual Age"
      postMenstrualAgeLabel="PMA"
      preferredFirstName={null}
    />
  }
/>
`;

exports[`renders the banner appropriately for a person with gestational and post menstrual dates 1`] = `
<ResponsiveElement
  responsiveTo="parent"
  small={
    <_default
      age="25 Years"
      applicationContent={
        <span
          className="risk-score"
        >
          5%
        </span>
      }
      dateOfBirth="May 9, 1993"
      dateOfBirthFullText="Date of Birth"
      dateOfBirthLabel="DOB"
      deceasedDate={null}
      deceasedDateLabel="Deceased"
      gender="Male"
      gestationalAge="April 5, 2016"
      gestationalAgeFullText="Gestational Age"
      gestationalAgeLabel="GA"
      identifiers={
        Object {
          "MRN": 12343,
          "REA": "3JSDA",
        }
      }
      isConfidential={false}
      personName="Johnathon Doe"
      photo=""
      postMenstrualAge="April 7, 2016"
      postMenstrualAgeFullText="Post Menstrual Age"
      postMenstrualAgeLabel="PMA"
      preferredFirstName="John"
    />
  }
  tiny={
    <_default
      age="25 Years"
      applicationContent={
        <span
          className="risk-score"
        >
          5%
        </span>
      }
      dateOfBirth="May 9, 1993"
      dateOfBirthFullText="Date of Birth"
      dateOfBirthLabel="DOB"
      deceasedDate={null}
      deceasedDateLabel="Deceased"
      gender="Male"
      gestationalAge="April 5, 2016"
      gestationalAgeFullText="Gestational Age"
      gestationalAgeLabel="GA"
      identifiers={
        Object {
          "MRN": 12343,
          "REA": "3JSDA",
        }
      }
      isConfidential={false}
      personName="Johnathon Doe"
      photo=""
      postMenstrualAge="April 7, 2016"
      postMenstrualAgeFullText="Post Menstrual Age"
      postMenstrualAgeLabel="PMA"
      preferredFirstName="John"
    />
  }
/>
`;

exports[`renders the banner properly for a deceased person with additional application content 1`] = `
<ResponsiveElement
  responsiveTo="parent"
  small={
    <_default
      age="25 Years"
      applicationContent={
        <span
          className="risk-score"
        >
          5%
        </span>
      }
      dateOfBirth="May 9, 1993"
      dateOfBirthFullText="Date of Birth"
      dateOfBirthLabel="DOB"
      deceasedDate="March 5, 2016"
      deceasedDateLabel="Deceased"
      gender="Male"
      gestationalAge={null}
      gestationalAgeFullText="Gestational Age"
      gestationalAgeLabel="GA"
      identifiers={
        Object {
          "MRN": 12343,
          "REA": "3JSDA",
        }
      }
      isConfidential={false}
      personName="Johnathon Doe"
      photo=""
      postMenstrualAge={null}
      postMenstrualAgeFullText="Post Menstrual Age"
      postMenstrualAgeLabel="PMA"
      preferredFirstName="John"
    />
  }
  tiny={
    <_default
      age="25 Years"
      applicationContent={
        <span
          className="risk-score"
        >
          5%
        </span>
      }
      dateOfBirth="May 9, 1993"
      dateOfBirthFullText="Date of Birth"
      dateOfBirthLabel="DOB"
      deceasedDate="March 5, 2016"
      deceasedDateLabel="Deceased"
      gender="Male"
      gestationalAge={null}
      gestationalAgeFullText="Gestational Age"
      gestationalAgeLabel="GA"
      identifiers={
        Object {
          "MRN": 12343,
          "REA": "3JSDA",
        }
      }
      isConfidential={false}
      personName="Johnathon Doe"
      photo=""
      postMenstrualAge={null}
      postMenstrualAgeFullText="Post Menstrual Age"
      postMenstrualAgeLabel="PMA"
      preferredFirstName="John"
    />
  }
/>
`;

exports[`renders the banner that contains additional information 1`] = `
<ResponsiveElement
  responsiveTo="parent"
  small={
    <_default
      age="25 Years"
      applicationContent={
        <span
          className="risk-score"
        >
          5%
        </span>
      }
      avatar={
        Object {
          "alt": "Johnathon Doe",
          "color": "neutral",
          "hashValue": "alternative hash",
          "image": "",
          "initials": "JD",
          "isAriaHidden": true,
        }
      }
      dateOfBirth="May 9, 1993"
      dateOfBirthFullText="Date of Birth"
      dateOfBirthLabel="DOB"
      deceasedDate={null}
      deceasedDateLabel="Deceased"
      gender="Male"
      gestationalAge={null}
      gestationalAgeFullText="Gestational Age"
      gestationalAgeLabel="GA"
      identifiers={
        Object {
          "MRN": 12343,
          "REA": "3JSDA",
        }
      }
      isConfidential={false}
      personName="Johnathon Doe"
      photo=""
      postMenstrualAge={null}
      postMenstrualAgeFullText="Post Menstrual Age"
      postMenstrualAgeLabel="PMA"
      preferredFirstName="John"
    />
  }
  tiny={
    <_default
      age="25 Years"
      applicationContent={
        <span
          className="risk-score"
        >
          5%
        </span>
      }
      avatar={
        Object {
          "alt": "Johnathon Doe",
          "color": "neutral",
          "hashValue": "alternative hash",
          "image": "",
          "initials": "JD",
          "isAriaHidden": true,
        }
      }
      dateOfBirth="May 9, 1993"
      dateOfBirthFullText="Date of Birth"
      dateOfBirthLabel="DOB"
      deceasedDate={null}
      deceasedDateLabel="Deceased"
      gender="Male"
      gestationalAge={null}
      gestationalAgeFullText="Gestational Age"
      gestationalAgeLabel="GA"
      identifiers={
        Object {
          "MRN": 12343,
          "REA": "3JSDA",
        }
      }
      isConfidential={false}
      personName="Johnathon Doe"
      photo=""
      postMenstrualAge={null}
      postMenstrualAgeFullText="Post Menstrual Age"
      postMenstrualAgeLabel="PMA"
      preferredFirstName="John"
    />
  }
/>
`;
